api/rs/slint/mcu.md

```diff
@@ -0,0 +1,405 @@
+   or from interrupt handlers.
+ * `libm`: MCUs often don't provide hardware support for floating point arithmetic. The `libm` feature enables the use of software emulation, with the help of
+   the [libm](https://crates.io/crates/libm) crate.
+
+Finally, we add `slint-build` as a build dependency in order to compile the `.slint` design files to Rust code.
+
+## Changes to `build.rs`
+
+When targeting MCUs, you need a [build script](https://doc.rust-lang.org/cargo/reference/build-scripts.html) to compile the `.slint` files using the `slint-build` crate.
+Use the `slint_build::EmbedResourcesKind::EmbedForSoftwareRenderer` configuration option to tell the slint compiler to embed the images and fonts in the binary
+in a format that's suitable for the software based renderer we're going to use.
+
+The following example of a `build.rs` script compiles the `main.slint` design file in the `ui/` sub-directory to Rust code and embeds the code as well as all
+graphical assets needed into the program binary.
+
+```rust,no_run
+fn main() {
+    slint_build::compile_with_config(
+        "ui/main.slint",
+        slint_build::CompilerConfiguration::new()
+            .embed_resources(slint_build::EmbedResourcesKind::EmbedForSoftwareRenderer),
+    ).unwrap();
+}
+```
+
+Hint: If you observe that your `build.rs` script is not used, double check that your `Cargo.toml` is using the
+[package.build](https://doc.rust-lang.org/cargo/reference/manifest.html#package-build) manifest key: `build = "build.rs"`.
+
+## Application Structure
+
+A graphical application in hosted environments is typically composed of at least three different tasks:
+
+ * Receive user input from operation system APIs.
+ * React to the input by performing application specific computations.
+ * Render an updated user interface and present it on the screen using device-independent operating system APIs.
+
+The operating system provides what is typically called an event loop to connect and schedule these tasks. Slint implements the
+task of receiving user input and forwarding it to the user interface layer, as well as rendering the interface to the screen.
+
+In bare metal environments it becomes your responsibility to substitute and connect functionality that is otherwise provided by the operating system:
+
+ * You need to select crates that allow you to initialize the chips that drive peripherals, such as a touch input or display controller.
+   Sometimes it may be necessary for you to develop your own drivers.
+ * You need to drive the event loop yourself by querying peripherals for input, forwarding input into computational modules of your
+   application and instructing Slint to render the user interface.
+
+In Slint the two primary APIs you need to use to accomplish these tasks the `Platform` trait as well as the `slint::Window` struct.
```

Comment:
`slint::platform::Platform`

Or 

just `Window`?

Reply:
Good point, that's inconsistent. I'll use their qualified names and make them links.